libraries/Microsoft.Bot.Builder.Ai.QnA/QnAMakerMiddleware.cs

```diff
@@ -56,5 +61,18 @@ public QnAMakerMiddleware(QnAMakerMiddlewareOptions options, HttpClient httpClie
 
             await next().ConfigureAwait(false);
         }
+
+        public static QnAMakerOptions RemoveSensitiveData(QnAMakerOptions options)
```

Comment:
This is strange. Seems like you could just code it out of existance. 

```
public class QnAMakerTraceRecord
{
  // does not have subscription key or KB Id. 
  // all "red flag" code around scrubbing and scensitive data would then go away. 
}
```

Reply:
Well, I was reusing the QnAMakerOptions class, which needs those attributes. I suppose I could move the attributes we need to the TraceInfo class. I was again copying the way that Luis does this.

Reply:
@cleemullins I moved the attributes from QnAMakerOptions out to QnAMakerTraceInfo, except for SubscriptionKey (KnowledgeBaseID is actually public, I made a mistake earlier by obfuscating it). What I liked about the previous way (obfuscating just the SubKey) is that it's more explicit about what we don't want to share. The way I have it now feels too much like cope paste coding.